pydbus/strong_typing.py

```diff
@@ -34,3 +35,10 @@ def decorate(func):
 		verify_arguments(func)
 		return func
 	return decorate
+
+
+class TypedSignal(signal):
+
+	def __init__(self, arguments):
```

Comment:
Hm... Realistic example would look like that:
```python
PropertiesChanged = typed_signal(("interface_name", "s"), ("changed_properties", "a{sv}"), ("invalidated_properties", "as"))
```
Not the prettiest thing ever, but I don't think we can find anything better with this syntax.

However, this one looks awesome (and Python3-only, so the previous solution is useful as a fallback):
```python
@signal
def PropertiesChanged(interface_name: 's', changed_properties: 'a{sv}', invalidated_properties: 'as'):
	pass
```
(Note, it can be done with `pydbus.generic.signal`, there is no need for another class here)

(2nd note: Sorry for constantly changing my opinion :D I can add the prettier syntax myself after merging the patch if you prefer :D)